base/applications/rapps/applicationdb.cpp

```diff
@@ -0,0 +1,292 @@
+/*
+ * PROJECT:     ReactOS Applications Manager
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Classes for working with available applications
+ * COPYRIGHT:   Copyright 2017 Alexander Shaposhnikov     (sanchaez@reactos.org)
+ *              Copyright 2020 He Yang                    (1160386205@qq.com)
+ *              Copyright 2021-2023 Mark Jansen <mark.jansen@reactos.org>
+ */
+
+#include "rapps.h"
+#include "applicationdb.h"
+#include "configparser.h"
+#include "settings.h"
+
+
+static HKEY g_RootKeyEnum[3] = {HKEY_CURRENT_USER, HKEY_LOCAL_MACHINE, HKEY_LOCAL_MACHINE};
+static REGSAM g_RegSamEnum[3] = {KEY_WOW64_32KEY, KEY_WOW64_32KEY, KEY_WOW64_64KEY};
+#define UNINSTALL_SUBKEY L"Software\\Microsoft\\Windows\\CurrentVersion\\Uninstall"
+
+static VOID ClearList(CAtlList<CApplicationInfo *>& list)
+{
+    POSITION InfoListPosition = list.GetHeadPosition();
+    while (InfoListPosition)
+    {
+        CApplicationInfo *Info = list.GetNext(InfoListPosition);
+        delete Info;
+    }
+    list.RemoveAll();
+}
+
+
+CApplicationDB::CApplicationDB(const CStringW& path)
+    :m_BasePath(path)
+{
+    m_BasePath.Canonicalize();
+}
+
+CApplicationInfo* CApplicationDB::FindByPackageName(const CStringW& name)
+{
+    POSITION CurrentListPosition = m_Available.GetHeadPosition();
+    while (CurrentListPosition)
+    {
+        CApplicationInfo* Info = m_Available.GetNext(CurrentListPosition);
+        if (Info->szIdentifier == name)
+        {
+            return Info;
+        }
```

Comment:
I do still think that such curly braces are nonsense if both: the command and the condition are both in a single line only.
Each time you do follow that antipattern you do waste 2 vertical lines of codes visibility on your monitor. And you waste in bytes in the source file 2 + 2x(amount of indentation space) bytes. Sorry, I just had to comment on that at least once for the many cases you do that.